pkg/networking/reconciler/reconciler.go

```diff
@@ -0,0 +1,38 @@
+package reconciler
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/controller"
+)
+
+type CiliumReconciler interface {
+	Reconcile(ctx context.Context, logger logr.Logger, client client.Client, spec *cluster.Spec) (controller.Result, error)
+}
+
+type Reconciler struct {
+	ciliumReconciler CiliumReconciler
+}
+
+func New(ciliumReconciler CiliumReconciler) *Reconciler {
+	return &Reconciler{
+		ciliumReconciler: ciliumReconciler,
+	}
+}
+
+// Reconcile takes the specified CNI in a cluster to the desired state defined in a cluster Spec
+// It uses a controller.Result to indicate when requeues are needed
+// Intended to be used in a kubernetes controller
+// Only Cilium CNI is supported for now
+func (r *Reconciler) Reconcile(ctx context.Context, logger logr.Logger, client client.Client, spec *cluster.Spec) (controller.Result, error) {
+	if spec.Cluster.Spec.ClusterNetwork.CNIConfig.Cilium != nil {
+		return r.ciliumReconciler.Reconcile(ctx, logger, client, spec)
+	} else {
+		return controller.Result{}, errors.New("unsupported CNI, only Cilium is supported at this time")
```

Comment:
When we return this error, the controller will requeue the request to be processed again, should we return nil err here?

Reply:
Great question. I'm on the fence, because this line is mostly for correctness, I'm not sure that this can happen in the real world, since the cluster won't pass the validations.

Maybe we can just write this error to the cluster status and return nil? Let's take this offline.

Reply:
The next time that it would get reconciled however is when there is a change to the cluster resource, or would the snow reconciler always invoke this reconciler anyways? Just want to clarify the trigger here

Reply:
The snow cluster reconciler would get called when the cluster object changes. And that reonciler would call the CNI one.